builtin/game/auth.lua

```diff
@@ -92,8 +92,15 @@ core.builtin_auth_handler = {
 
 		core_auth.save(auth_entry)
 
-		-- Run grant callbacks
-		for priv, _ in pairs(privileges) do
+		for priv, value in pairs(privileges) do
+			-- Warnings for improper API usage
+			if value == false then
+				core.log('warning', "`false` value in `privs`, this is almost certainly a bug, "..
```

Comment:
should probably be `core.log("deprecated"`

also I would leave off the `"granting a privilege rather than revoking it"` part

Reply:
+1 for making this a deprecation warning, i.e. one that triggers an error if `deprecated_lua_api_handling = error`. (Same for the the other warnings.)

I find the `granting a privilege rather than revoking it` part helpful.

To have the API function name in the error message, I'd write ...

```
`false` value given to `minetest.set_player_privs`
```

... instead of ...

```
`false` value in `privs`
```

(Same for the other warnings.)

Reply:
I went with grorp's suggestion.